test/jit/test_pdt.py

```diff
@@ -90,3 +90,204 @@ def fun(self, x):
         scripted_pdt_model = torch.jit._script_pdt(pdt_model, example_inputs=[(20, ), (False, )])
         self.assertEqual(scripted_pdt_model(30), pdt_model(30))
         self.assertEqual(scripted_pdt_model(True), pdt_model(True))
+    def test_pdt(self):
+        def test_sum(a, b):
+            return a + b
+
+        make_global(test_sum)
+        scripted_fn_add = torch.jit._script_pdt(test_sum, example_inputs=[(3, 4)])
+        self.assertEqual(scripted_fn_add(10, 2), test_sum(10, 2))
+
+        def test_sub(a, b):
+            return a - b
+
+        make_global(test_sub)
+        scripted_fn_sub = torch.jit._script_pdt(test_sub, example_inputs=[(3.9, 4.10)])
+        self.assertEqual(scripted_fn_sub(6.5, 2.9), test_sub(6.5, 2.9))
+
+        def test_mul(a, b):
+            return a * b
+
+        make_global(test_mul)
+        scripted_fn_mul = torch.jit._script_pdt(test_mul, example_inputs=[(-10, 9)])
+        self.assertEqual(scripted_fn_mul(-1, 3), test_mul(-1, 3))
+
+        def test_args_complex(real, img):
+            return torch.complex(real, img)
+
+        make_global(test_args_complex)
+        scripted_fn_complex = torch.jit._script_pdt(test_args_complex, example_inputs=[(torch.rand(3, 4), torch.rand(3, 4))])
+        arg1, arg2 = torch.rand(3, 4), torch.rand(3, 4)
+        self.assertEqual(scripted_fn_complex(arg1, arg2), test_args_complex(arg1, arg2))
+
+        def test_bool(a):
+            if a:
+                return -1
+            else:
+                return 0
+
+        make_global(test_bool)
+        scripted_fn_bool = torch.jit._script_pdt(test_bool, example_inputs=[(True,)])
+        self.assertEqual(scripted_fn_bool(True), test_bool(True))
+
+        def test_str(a):
+            if a == "":
+                return False
+            else:
+                return True
+
+        make_global(test_str)
+        scripted_fn_str = torch.jit._script_pdt(test_str, example_inputs=[("",)])
+        self.assertEqual(scripted_fn_str("abc"), test_str("abc"))
+
+    def test_pdt_list_and_tuple(self):
+        def test_list_and_tuple(a):
+            return sum(a)
+
+        make_global(test_list_and_tuple)
+
+        scripted_fn_float = torch.jit._script_pdt(test_list_and_tuple, example_inputs=[([4.9, 8.9],)])
+        self.assertEqual(scripted_fn_float([11.9, 7.6]), test_list_and_tuple([11.9, 7.6]))
+
+        scripted_fn_bool = torch.jit._script_pdt(test_list_and_tuple, example_inputs=[([True, False, True],)])
+        self.assertEqual(scripted_fn_bool([True, True, True]), test_list_and_tuple([True, True, True]))
+
+        scripted_fn_int = torch.jit._script_pdt(test_list_and_tuple, example_inputs=[([3, 4, 5], )])
+        self.assertEqual(scripted_fn_int([1, 2, 3]), test_list_and_tuple([1, 2, 3]))
+
+        scripted_fn_float = torch.jit._script_pdt(test_list_and_tuple, example_inputs=[((4.9, 8.9),)])
+        self.assertEqual(scripted_fn_float((11.9, 7.6)), test_list_and_tuple((11.9, 7.6)))
```

Comment:
FYI @ansley 

These tests actually reminds me of a next step feature we need:

When combining types like `List[int]` and `List[str]`, maybe we can generate `List[Any]` instead of `Any`, so we retain a little more information about the type.

Reply:
Another thing I realized is that:
When we have generic function/method overloading support (@tugsbayasgalan is working on part of it now), maybe we don't need to combine the types any more in some cases, instead we just generate a few different overloads of method, this will be lower runtime overhead and can capture type relationships like `type(a) === type(b)` 

for example

```
def fn(a, b):
    return a + b
```

The intention of function may be that type of `a` and `b` are always the same. This is lost if we analyze types of both to be `Any`. However, this relationship will be kept if we generate two different overloads of `fn`:
```
def fn(a: int, b: int):
   ...

def fn(a: float, b: float):
  ...
```